scss/bitstyles/utilities/margin/margin.stories.mdx

```diff
@@ -4,46 +4,220 @@ import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+        <div class="u-bg--background u-margin-xs">u-margin-xs</div>
+      </div>
+    `}
+  </Story>
+  <Story name="margin-s">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-s">u-margin-s</div>
+      </div>
+    `}
+  </Story>
   <Story name="margin-m">
     {`
       <div class="u-bg--brand-2 u-padding-m">
         <div class="u-bg--background u-margin-m">u-margin-m</div>
       </div>
     `}
   </Story>
-  <Story name="margin-0">
+  <Story name="margin-l">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-l">u-margin-l</div>
+      </div>
+    `}
+  </Story>
+  <Story name="margin-xl">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-xl">u-margin-xl</div>
+      </div>
+    `}
+  </Story>
+</Canvas>
+
+## Negative margin sizes
+
+<Canvas>
+  <Story name="margin-neg-xxs">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-neg-xxs">u-margin-neg-xxs</div>
+      </div>
+    `}
+  </Story>
+  <Story name="margin-neg-xs">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-neg-xs">u-margin-neg-xs</div>
+      </div>
+    `}
+  </Story>
+  <Story name="margin-neg-s">
     {`
       <div class="u-bg--brand-2 u-padding-m">
-        <p class="u-bg--background u-margin-0">u-margin-0</p>
+        <div class="u-bg--background u-margin-neg-s">u-margin-neg-s</div>
       </div>
     `}
   </Story>
-  <Story name="margin-m--top">
+  <Story name="margin-neg-m">
     {`
       <div class="u-bg--brand-2 u-padding-m">
-        <div class="u-bg--background u-margin-m--top">u-margin-m--top</div>
+        <div class="u-bg--background u-margin-neg-m">u-margin-neg-m</div>
       </div>
     `}
   </Story>
-  <Story name="margin-m--right">
+  <Story name="margin-neg-l">
     {`
       <div class="u-bg--brand-2 u-padding-m">
-        <div class="u-bg--background u-margin-m--right">u-margin-m--right</div>
+        <div class="u-bg--background u-margin-neg-l">u-margin-neg-l</div>
       </div>
     `}
   </Story>
-  <Story name="margin-m--bottom">
+  <Story name="margin-neg-xl">
+    {`
+      <div class="u-bg--brand-2 u-padding-m">
+        <div class="u-bg--background u-margin-neg-xl">u-margin-neg-xl</div>
+      </div>
+    `}
+  </Story>
+</Canvas>
+
+## Directional margins
+
+Every margin size (negative and positive) is also available directed at the single sides `top`, `right`, `bottom`, `left`, and the double directions `x` and `y`.
+
+This can be customised by overriding `$bitstyles-margin-directions` e.g.
+
+```scss
+$bitstyles-margin-directions: (
+  '': null,
+  'top': ('top'),
+  'right': ('right'),
+  'bottom': ('bottom'),
+  'left': ('left'),
+  'x': ('right', 'left'),
+  'y': ('top', 'bottom')
+);
+```
+
+The `'': null` entry is needed to render the margin classes with no directional modifiers.
```

Comment:
Could this also be done with a default value for the function parameter? Plus: why is there no example for this?

Reply:
The examples for these (this and the padding one you flagged too) are all the examples that came before — the classes above all change the size of the margin, but are not directional (they apply to all directions). Since that’s not clear, and not having it _is_ inconsistent, I’ll add it as the first example in the following block of examples. Same for the padding. Thanks!

Reply:
Ok I added the all-directional examples here too ✨

Your other question about doing this with a default parameter: I wanted this to all be as configurable as possible, so that only classes you need are output (it’s common to avoid using margin-top, for example). It would be inconsistent to not allow the all-directional versions to be configured. But it did result in this slightly odd config 🤔 I’ll think on it, maybe your suggestion will work for all cases we need it to, or maybe there’s just a better way to do this

Reply:
I’m going to go with this approach for now, and reconsider when I know how we end up using these functions and configurations — many of the utility classes could be better-built using generators. Thanks for your feedback, it was super helpful!